packages/flutter/test/material/progress_indicator_test.dart

```diff
@@ -531,6 +531,29 @@ void main() {
     expect(find.byType(CircularProgressIndicator), paints..arc(strokeCap: StrokeCap.round));
   });
 
+  testWidgets('CircularProgressIndicator with custom animation controller', (
+    WidgetTester tester,
+  ) async {
+    late AnimationController controller1;
+
+    Widget buildWidget(AnimationController? controller) {
+      return MaterialApp(
+        home: Scaffold(body: Center(child: CircularProgressIndicator(value: controller?.value))),
+      );
+    }
+
+    controller1 = AnimationController(vsync: tester, duration: const Duration(seconds: 2));
+
+    await tester.pumpWidget(buildWidget(controller1));
+    expect(controller1.value, 0.0);
```

Comment:
Great so far! Can you test that the controller works? For example by making the animation move forward.